umake/frameworks/games.py

```diff
@@ -385,3 +386,70 @@ def post_install(self):
                         exec=self.exec_link_name,
                         comment=self.description,
                         categories="Development;IDE;"))
+
+
+class Godot(umake.frameworks.baseinstaller.BaseInstaller):
+
+    def __init__(self, **kwargs):
+        super().__init__(name="Godot",
+                         description=_("Multi-platform 2D and 3D opensource game engine"),
+                         only_on_archs=['i386', 'amd64'],
+                         download_page="https://godotengine.org/download",
+                         desktop_filename="godot.desktop",
+                         required_files_path=["godot"],
+                         dir_to_decompress_in_tarball="",
+                         icon_filename='icon.png',
+                         **kwargs)
+        self.icon_url = os.path.join("https://raw.githubusercontent.com/godotengine/godot/",
+                                     "master",
+                                     self.icon_filename)
+        self.exec_name = ''
+
+    arch_trans = {
+        "amd64": "64",
+        "i386": "32"
+    }
+
+    def parse_download_link(self, line, in_download):
+        """Parse Godot download links"""
+        url = None
+        if '.zip' in line:
+            p = re.search(r'href="([^<]*{}.zip)"'.format(self.arch_trans[get_current_arch()]), line)
+            with suppress(AttributeError):
+                url = p.group(1)
+        return ((url, None), in_download)
+
+    def decompress_and_install(self, fds):
+        """Get Godot executable name from url"""
+        super().decompress_and_install(fds)
+        exec_url = self.download_requests[0].url
+        self.exec_name = os.path.splitext(os.path.basename(exec_url))[0]
+
+    def save_icon(self, download_result):
+        """Save correct Godot icon"""
+        icon = download_result.pop(self.icon_url).fd.name
+        shutil.copy(icon, os.path.join(self.install_path, self.icon_filename))
+        logger.debug("Copied icon: {}".format(self.icon_url))
+
+    def post_install(self):
+        """Create the Godot Code launcher"""
+        DownloadCenter(urls=[DownloadItem(self.icon_url, None)],
+                       on_done=self.save_icon, download=True)
+
+        # Rename Godot executable
+        os.rename(os.path.join(self.install_path, self.exec_name),
+                  os.path.join(self.install_path, self.exec_link_name))
+
+        # Get icon path
+        icon_path = join(self.install_path, self.icon_filename)
+
+        comment = self.description
+        categories = "Development;IDE;"
```

Comment:
Just add self.description and the categories in the create launcher in the following lines.
Also, remove the empty lines